src/components/Account/AccountModule.js

```diff
@@ -1,145 +1,81 @@
-    height: 22 * GU,
-  },
-  {
-    id: 'error',
-    title: 'Connection error',
-    height: 50 * GU,
-  },
+  { id: 'providers', title: 'Use account from' },
+  { id: 'connecting', title: 'Use account from' },
+  { id: 'connected', title: 'Active account' },
+  { id: 'error', title: 'Connection error' },
 ]
 
 function AccountModule() {
-  const buttonRef = useRef()
-  const wallet = useWallet()
   const [opened, setOpened] = useState(false)
-  const [animate, setAnimate] = useState(false)
   const [activatingDelayed, setActivatingDelayed] = useState(false)
-  const [activationError, setActivationError] = useState(null)
-  const popoverFocusElement = useRef()
-
+  const buttonRef = useRef()
   const { below } = useViewport()
   const compactMode = below('medium')
+  const wallet = useWallet()
+  const { account, connector, error, status } = wallet
 
-  const { account, activating } = wallet
-
-  const clearError = useCallback(() => setActivationError(null), [])
-
+  const open = useCallback(() => setOpened(true), [])
   const toggle = useCallback(() => setOpened(opened => !opened), [])
 
-  const handleCancelConnection = useCallback(() => {
-    wallet.deactivate()
-  }, [wallet])
-
-  const activate = useCallback(
-    async providerId => {
-      try {
-        await wallet.activate(providerId)
-      } catch (error) {
-        setActivationError(error)
-      }
-    },
-    [wallet]
-  )
-
-  // Don’t animate the slider until the popover has opened
   useEffect(() => {
-    if (!opened) {
-      return
-    }
-    setAnimate(false)
-    const timer = setTimeout(() => {
-      setAnimate(true)
-    }, 0)
-    return () => clearTimeout(timer)
-  }, [opened])
+    let timer
 
-  // Always show the “connecting…” screen, even if there are no delay
-  useEffect(() => {
-    if (activationError) {
+    if (status === 'error') {
       setActivatingDelayed(null)
     }
 
-    if (activating) {
-      setActivatingDelayed(activating)
-      return
+    if (status === 'connecting') {
+      setActivatingDelayed(connector)
+      timer = setTimeout(() => {
+        setActivatingDelayed(null)
+      }, 400)
     }
 
-    const timer = setTimeout(() => {
-      setActivatingDelayed(null)
-    }, 500)
+    return () => clearTimeout(timer)
+  }, [connector, status])
 
-    return () => {
-      clearTimeout(timer)
-    }
-  }, [activating, activationError])
+  const handleCancelConnection = useCallback(() => {
+    wallet.reset()
+  }, [wallet])
+
+  const handleActivate = useCallback(providerId => wallet.connect(providerId), [
+    wallet,
+  ])
 
   const previousScreenIndex = useRef(-1)
 
   const { screenIndex, direction } = useMemo(() => {
-    const screenId = (() => {
-      if (activationError) return 'error'
-      if (activatingDelayed) return 'connecting'
-      if (account) return 'connected'
-      return 'providers'
-    })()
+    const screenId = (() =>
+      status === 'disconnected' ? 'providers' : status)()
```

Comment:
Should we remove the wrapped function now?

Reply:
Let's do it!